app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -363,12 +390,59 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
         }
     }
 
-    private fun updateAccountProfile(profile: Profile) {
+    private fun updateAccountProfile(profile: Profile, error: Boolean = false) {
         launch {
             val preferenceFirefoxAccount =
-                findPreference<Preference>(context!!.getPreferenceKey(pref_key_account))
-            preferenceFirefoxAccount?.title = profile.displayName.orEmpty()
-            preferenceFirefoxAccount?.summary = profile.email.orEmpty()
+                findPreference<AccountPreference>(context!!.getPreferenceKey(pref_key_account))
+
+            preferenceFirefoxAccount?.title?.setTextColor(
+                R.attr.primaryText.getColorFromAttr(context!!)
+            )
+            preferenceFirefoxAccount?.title?.text = profile.displayName.orEmpty()
+            preferenceFirefoxAccount?.title?.visibility =
+                if (preferenceFirefoxAccount?.title?.text.isNullOrEmpty()) View.GONE else View.VISIBLE
+
+            preferenceFirefoxAccount?.summary?.setTextColor(
+                R.attr.primaryText.getColorFromAttr(context!!)
+            )
+            preferenceFirefoxAccount?.summary?.text = profile.email.orEmpty()
+
+            preferenceFirefoxAccount?.icon = ContextCompat.getDrawable(context!!, R.drawable.ic_shortcuts)
+            preferenceFirefoxAccount?.errorIcon?.visibility = View.GONE
+            preferenceFirefoxAccount?.background?.background = null
+        }
+    }
+
+    private fun displayAccountErrorIfNecessary() {
+        if (!org.mozilla.fenix.utils.Settings.getInstance(context!!).hasSyncProblem) { return }
```

Comment:
It'd be more reliable if this method actually queried account state via `accountManager.accountNeedsReauth`